lib/shopify_app/controller_concerns/login_protection.rb

```diff
@@ -27,20 +27,38 @@ def activate_shopify_session
     end
 
     def current_shopify_session
-      if session[:user_id].present?
-        @current_shopify_session ||= user_session
-      else
-        @current_shopify_session ||= shop_session
+      @current_shopify_session ||= begin
+        user_session || shop_session
       end
     end
 
     def user_session
-      return if session[:user_id].blank?
+      user_session_by_jwt || user_session_by_cookie
+    end
+
+    def user_session_by_jwt
+      return unless ShopifyApp.configuration.allow_jwt_authentication
+      return unless jwt_shopify_user_id
+      ShopifyApp::SessionRepository.retrieve_user_session_by_shopify_user_id(jwt_shopify_user_id)
```

Comment:
Why do we call a different method here than we do on line 47 if it's a user id in both cases?

Reply:
So line 47 leverages a user id that is generated by `shopify_app` but `shopify_user_id` is different where it is known across all of shopify for the particular user.